### DISCUSSION TITLE
关于自定义 method key

### DISCUSSION THREAD

**Discussion by Ya-hui** [alovajs/alova#572] [Q&A]
**我看官网有这段介绍**

> 响应数据缓存的 key 是由 method 实例的请求方法(method)、请求地址(url)、请求头参数(headers)、url 参数(params)、请求体参数(requestBody)组合作为唯一标识，任意一个信息或位置不同都将被当做不同的 key，如果要自定义缓存 key，可以参考[自定义 method key](https://alova.js.org/zh-CN/tutorial/advanced/in-depth/custom-method-key)。

我们这边接口是有时间戳以及签名的， 所以每次请求应该都会生成一个新 key ， 所以不会触发缓存。 只能自定义缓存key ,  然后看文档可以这么写
```ts
const methodInst = alovaInstance.Get('/api/user', {});
methodInst.key = 'my-custom-method-key';
```
感觉这样太不够优雅了， 很多代码都是直接 await出去了 比如：

```ts
const methodInst = await alovaInstance.Get('/api/user', {});
```

然后还可以这样

```ts
import { Method } from 'alova';

Method.prototype.generateKey = function () {
  console.log(this.context)
  return 'your-custom-method-key';
};

```

我感觉这种方式比较适合我，把签名和时间戳 删除， 然后根据 请求方法(method)、请求地址(url)、请求头参数(headers)、url 参数(params)、请求体参数(requestBody)组合作为唯一标识。  但是我打印this.context， 并拿不到一些请求信息， 比如请求url 请求参数 ， 没有这些信息 我怎么给请求方法 标识一个唯一的key呢？ 
![image](https://github.com/user-attachments/assets/94a5af1d-62f3-487b-88d6-4522d6bb988e)

**Comment by JOU-amjs** [comment]
你打印`this`看看就知道了

**Reply by Ya-hui** [reply]
很棒， 还有个问题 全局设置的createAlova({cacheFor : {}})会和GET(‘’，{cacheFor: {}})合并吗？

**Reply by JOU-amjs** [reply]
method设置了则以method为准，否则以createAlova为准

**Reply by Ya-hui** [reply]
tag 标签还需要我自己生成key的时候 维护吗？

**Reply by Ya-hui** [reply]
我在里面打印this 会进来两次
console.log(this)

第一次进来是为了检查缓存？ 第二次不存在设置？ 因为这两次的this 参数不一样 。 第一次看着比较像原始参数， 第二次参数都是经过格式化的了

**Reply by JOU-amjs** [reply]
是的，会在调用usehooks时先检查缓存，第二次是发送请求的调用
